sympy/matrices/matrices.py

```diff
@@ -3825,8 +3825,11 @@ def pinv(self):
                 return AH * (A * AH).inv()
         except ValueError:
             # Matrix is not full rank, so A*AH cannot be inverted.
-            raise NotImplementedError('Rank-deficient matrices are not yet '
-                                      'supported.')
+            P, D = (A.H * A).diagonalize()
+            for i in range(P.cols):
+                P[:,i] = P.col(i) / P.col(i).norm()
```

Comment:
Use index notation instead of `.col(i)`

Reply:
While checking out diagonalize(), I just noticed that it has a `normalize` parameter that will do the normalization for us. So I'll use that instead and delete this for-loop.